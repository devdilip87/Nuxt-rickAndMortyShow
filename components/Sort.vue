<template>
    <div class="sort">
        <select v-model="defaultSortOption">
            <option v-for="(option, index) in sortOptions" :value="index" :disabled="index == defaultSortOption" v-bind:key="index">{{option}}</option>
            
        </select>
    </div>
</template>

<script>
export default {
  data () {
    return {
      defaultSortOption: 0,
      sortOptions: ['Sort by ID', 'Ascending', 'Decending']
    }
  },
  watch: {
    defaultSortOption: function(n, o) {
      if(n && n !== o) {
        const payload = {
            sortOption: this.defaultSortOption,
            list: this.$store.state.filteredList
        }
        this.$store.dispatch("sortList", payload);
      }
    }
  }
}
</script>

<style>
</style>
