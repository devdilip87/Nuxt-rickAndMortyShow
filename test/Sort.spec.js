import { mount, shallowMount } from '@vue/test-utils'
import Sort from '@/components/Sort.vue'

describe('Sort', () => {
    test('is a Vue instance', () => {
        const wrapper = shallowMount(Sort)
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    test('it update default option', () => {
        const wrapper = shallowMount(Sort)
        wrapper.vm.$data.defaultSortOption = 1
        expect(wrapper.vm.$data.defaultSortOption).toBe(1)
    })

    test('it generate snapshot test', () => {
        const wrapper = shallowMount(Sort)
        expect(wrapper.html()).toMatchSnapshot()
    })
})