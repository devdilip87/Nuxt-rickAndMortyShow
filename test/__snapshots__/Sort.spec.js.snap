// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort it generate snapshot test 1`] = `
"<div class=\\"sort\\"><select>
    <option disabled=\\"disabled\\" value=\\"0\\">Sort by ID</option>
    <option value=\\"1\\">Ascending</option>
    <option value=\\"2\\">Decending</option>
  </select></div>"
`;
